src/inference/hmc.jl

```diff
@@ -100,21 +100,52 @@ function HMC{AD}(
     return HMC{AD}(ϵ, n_leapfrog, metricT, space)
 end
 
+function update_hamiltonian!(spl, model, n)
+    metric = gen_metric(n, spl)
+    ℓπ = gen_logπ(spl.state.vi, spl, model)
+    ∂ℓπ∂θ = gen_∂logπ∂θ(spl.state.vi, spl, model)    
+    spl.state.h = AHMC.Hamiltonian(metric, ℓπ, ∂ℓπ∂θ)
+    return spl
+end
+
 function AbstractMCMC.sample_init!(
     rng::AbstractRNG,
     model::AbstractModel,
     spl::Sampler{<:Hamiltonian},
     N::Integer;
     verbose::Bool=true,
     resume_from=nothing,
+    init_theta=nothing,
     kwargs...
 )
-
     # Resume the sampler.
     set_resume!(spl; resume_from=resume_from, kwargs...)
 
     # Get `init_theta`
     initialize_parameters!(spl; verbose=verbose, kwargs...)
+    if init_theta !== nothing
+        # Doesn't support dynamic models
+        link!(spl.state.vi, spl)
```

Comment:
I would assume `init_theta` is in the constrained space. But here we move to unconstrained space?

Reply:
It is in the constrained space so we transform it to the unconstrained. `initialize_parameters!` puts `init_theta` in `vi` so no we transform it here with `link!`. Here we assume the trans fields in `vi` are all false, but we probably need to check that before calling `initialize_parameters!`.